addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -25,3 +30,202 @@ def _compute_proxy_type(self):
         for user in self:
             if user.company_id.country_code == 'IT':
                 user.proxy_type = 'l10n_it_edi'
+
+    def _get_proxy_identification(self, company, proxy_type):
+        if proxy_type == 'l10n_it_edi':
+            if not company.l10n_it_codice_fiscale:
+                raise UserError(_('Please fill your codice fiscale to be able to receive invoices from FatturaPA'))
+            return self.env['res.partner']._l10n_it_edi_normalized_codice_fiscale(company.l10n_it_codice_fiscale)
+        return super()._get_proxy_identification(company, proxy_type)
+
+    def _l10n_it_edi_upload(self, files):
+        '''Upload files to the SdI.
+
+        :param files:    A list of dictionary {filename, base64_xml}.
+        :returns:        A dictionary.
+        * message:       Message from fatturapa.
+        * transactionId: The fatturapa ID of this request.
+        * error:         An eventual error.
+        * error_level:   Info, warning, error.
+        '''
+        if not files:
+            return {}
+        if self.edi_mode == 'demo':
+            return {file_data['filename']: {'id_transaction': 'demo'} for file_data in files}
+
+        ERRORS = {'EI01': _('Attached file is empty'),
+                  'EI02': _('Service momentarily unavailable'),
+                  'EI03': _('Unauthorized user')}
+
+        server_url = self._get_server_url()
+        results = self._make_request(
+            f'{server_url}/api/l10n_it_edi/1/out/SdiRiceviFile',
+            params={'files': files})
+
+        for filename, vals in results.items():
+            if 'error' in vals:
+                results[filename]['error'] = ERRORS.get(vals.get('error'), _("Unknown error"))
+
+        return results
+
+    def cron_l10n_it_edi_download_and_update(self):
+        """ Crons run with sudo(), with empty recordset. Remember that. """
+        retrigger = False
+        for proxy_user in self.search([('proxy_type', '=', 'l10n_it_edi')]):
+            user = proxy_user.with_company(proxy_user.company_id)
+            if user.edi_mode != 'demo':
+                user._l10n_it_edi_update_send_state()
+                retrigger = retrigger or user._l10n_it_edi_download()
+            else:
+                user._l10n_it_edi_demo_mode_update()
```

Comment:
To me, having all that code on the proxy user is weird. The proxy user is the way to communicate but that's used only in invoices so everything should be on account.move, in a dedicated section and well prefixed by `_l10n_it_edi_`